cloud-service/src/login/mod.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Golem Source License v1.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://license.golem.cloud/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod oauth2;
+mod oauth2_github_client;
+mod oauth2_provider_client;
+mod oauth2_session;
+mod oauth2_token_repo;
+mod oauth2_web_flow_state_repo;
+mod service;
+
+pub use self::oauth2::OAuth2Error;
+use self::oauth2::OAuth2Service;
+pub use self::oauth2_token_repo::{DbOAuth2TokenRepo, OAuth2TokenRepo};
+pub use self::oauth2_web_flow_state_repo::{DbOAuth2FlowState, OAuth2WebFlowStateRepo};
+pub use self::service::LoginError;
+pub use self::service::LoginService;
+use crate::config::LoginConfig;
+use crate::service::account::AccountService;
+use crate::service::token::TokenService;
+use golem_service_base::db::Pool;
+use std::sync::Arc;
+
+pub struct LoginSystemEnabled {
+    pub login_service: Arc<dyn LoginService>,
+    pub oauth2_service: Arc<dyn OAuth2Service>,
+}
+
+pub enum LoginSystem {
+    Enabled(LoginSystemEnabled),
```

Comment:
All the oauth2 / login stuff is gated behind this.

Optimally we would not even migrate the tables if this is disabled, but it's not worth the effort to implement currently imo